libass/ass_directwrite.c

```diff
@@ -253,79 +253,92 @@ static void init_FallbackLogTextRenderer(FallbackLogTextRenderer *r,
     };
 }
 
+/*
+ * A helper function for init_font_private_stream and get_utf8_path.
+ * Remember to release the file and loader after obtaining it, even if the function fails.
+ */
+static bool get_font_file_loader_and_reference_key(
+    FontPrivate *priv,
+    const void **refKey,
+    UINT32* keySize,
+    IDWriteFontFile **file,
+    IDWriteFontFileLoader **loader
+    )
```

Comment:
That too, but I suspect MrSmile means the way the whole declaration is split across lines. Usually, we do it like this:

```c
static bool get_font_file_loader_and_reference_key(FontPrivate *priv,
                                                   ...,
                                                   IDWriteFontFileLoader **loader)
{
```

But once again, this file already breaks with this tradition for some of its functions, and you’ve merely copied that second style here. It looks like it does this for functions with super long names, where fitting even one parameter on the same line as the function name makes the line arguably too long, e. g.:

https://github.com/libass/libass/blob/6a759836e5e76bb7b69b0ac244eea76b0d290512/libass/ass_directwrite.c#L115-L120

Personally, I don’t have a strong opinion on this, but perhaps try using our usual style, and if it you find the lines to be too long, try shortening the function name? I do feel it’s pretty verbose. You could perhaps do with `get_loader_and_reference` or `get_font_reference` or `get_loader_reference` or `get_font_loader` or maybe even something completely different.

Reply:
Yeah between lines 96 and 217 all the funcs are in this style and I must have mimicked it from there.

I fixed it up to match up the style. As for names I have a few more ideas, but then again the function is pretty specific and does all these things grabbing these objects at once, so perhaps it's not that bad.

It could've been worse like `get_font_file_and_font_file_loader_and_reference_key` or something xD

But currently it's named for the two things we actually *need* from the function, being the font file loader and reference key. The third item font file we grab just to properly dereference it later, so it's not mentioned as the intermediary there. I think it kinda makes sense.

At most it could be shortened to `get_font_loader_and_reference_key`, since there can't be non-file loader, but then again emphasizing "file" makes sense if one of the funcs has to obtain the file's path. So yeah I'd just keep it...